packages/rum-core/src/domain/action/getActionNameFromElement.ts

```diff
@@ -176,11 +177,15 @@ function getActionNameFromElementForStrategies(
     element.nodeName !== 'HEAD'
   ) {
     for (const strategy of strategies) {
-      const name = strategy(element, userProgrammaticAttribute, privacyEnabledActionName)
-      if (typeof name === 'string') {
+      const actionName = strategy(element, userProgrammaticAttribute, privacyEnabledActionName)
+      if (actionName) {
+        const { name, nameSource } =
+          typeof actionName === 'string'
+            ? { name: actionName, nameSource: ActionNameSource.STANDARD_ATTRIBUTE }
+            : actionName
```

Comment:
**💬 suggestion:** ‏I think it would be simpler if strategies didn't return `string`. If you want to factorize attributes, you could have a function like:
```js
function getActionNameFromStandardAttribute(element, attribute) {
  return {
    name: element.getAttribute(attribute),
    nameSource: ActionNameSource.STANDARD_ATTRIBUTE
  }
}